new-implementation/aws/backends/eks/main.tf

```diff
@@ -0,0 +1,88 @@
+//local variables 
+
+locals {
+  instance_create_timeout = "60"
+}
+
+data "aws_availability_zones" "available" {
+}
+
+data "aws_eks_cluster" "cluster" {
+  name = module.eks.cluster_id
+}
+
+data "aws_eks_cluster_auth" "cluster" {
+  name = module.eks.cluster_id
+}
+
+
+data "template_file" "public_cidrs" {
+  count = 3
+  template = "${cidrsubnet("10.0.0.0/16", 8, count.index )}"
+}
+
+data "template_file" "database_cidrs" {
+  count = 2
+  template = "${cidrsubnet("10.0.0.0/16", 8, count.index + 3)}"
+}
+
+// ssh keypair
+resource "aws_key_pair" "keypair" {
+  key_name = var.cluster_name
+  public_key = file(var.public_key_path)
+}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "2.6.0"
+
+  name                 = var.cluster_name
+  cidr                 = "10.0.0.0/16"
+  azs                  = data.aws_availability_zones.available.names
+  database_subnets = "${data.template_file.database_cidrs.*.rendered}"
+  public_subnets = "${data.template_file.public_cidrs.*.rendered}"
+  enable_dns_hostnames = true
+  enable_nat_gateway = true
+  single_nat_gateway = true
+  tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
```

Comment:
implementation wise I think it may make sense to have each of the backends be as independent as possible, so if someone just wants eks or if someone just wants rds, they can have those?  When we combine them, instead of the postgres making its vpc for its subnet, the vpc would be the same as the one as eks, but a subnet inside the vpc would get created?